samples/app/src/androidTest/java/com/sourcepoint/app/v6/UserConsentsRegressionTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.sourcepoint.app.v6
+
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import androidx.test.platform.app.InstrumentationRegistry
+import com.example.uitestutil.assertEquals
+import com.example.uitestutil.toList
+import com.example.uitestutil.wr
+import com.sourcepoint.cmplibrary.SpClient
+import com.sourcepoint.cmplibrary.creation.config
+import com.sourcepoint.cmplibrary.exception.CampaignType
+import com.sourcepoint.cmplibrary.model.MessageLanguage
+import com.sourcepoint.cmplibrary.util.userConsents
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import org.json.JSONObject
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.core.context.loadKoinModules
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+class UserConsentsRegressionTest {
+
+    private val appContext by lazy { InstrumentationRegistry.getInstrumentation().targetContext }
+
+    private val spConfGdpr = config {
+        accountId = 22
+        propertyId = 16893
+        propertyName = "mobile.multicampaign.demo"
+        messLanguage = MessageLanguage.ENGLISH
+        +(CampaignType.GDPR)
+    }
+
+    @Test
+    fun given_the_user_provides_partial_consent_for_gdpr_then_user_consents_should_return_proper_consent() = runBlocking<Unit> {
+
+        val v7Consent = JSONObject(TestData.storedConsentV741)
```

Comment:
If I'm not mistaken those data have AcceptAll, you can add a new test data with "SaveAndExit"
`
{
  "sp.ccpa.key.consent.status": "{\n  \"applies\": true,\n  \"consentedAll\": false,\n  \"dateCreated\": \"2023-11-07T14:10:06.932Z\",\n  \"rejectedAll\": false,\n  \"rejectedCategories\": [\n    \"608bae685461ff11a2c28658\"\n  ],\n  \"rejectedVendors\": [\n    \"5dcb0978fcf9312bfc52ae91\"\n  ],\n  \"signedLspa\": false,\n  \"uspstring\": \"1YYN\",\n  \"status\": \"rejectedSome\",\n  \"uuid\": \"7014c66c-1316-4644-aa46-3d85292127a1\",\n  \"webConsentPayload\": {\n    \"actions\": [\n    ],\n    \"cookies\": [\n      {\n        \"key\": \"ccpaUUID\",\n        \"value\": \"7014c66c-1316-4644-aa46-3d85292127a1\",\n        \"setPath\": true,\n        \"maxAge\": 31536000\n      },\n      {\n        \"key\": \"ccpaReject\",\n        \"value\": true,\n        \"setPath\": true,\n        \"maxAge\": 31536000\n      },\n      {\n        \"key\": \"ccpaConsentAll\",\n        \"value\": false,\n        \"setPath\": true,\n        \"maxAge\": 31536000\n      },\n      {\n        \"key\": \"consentStatus\",\n        \"value\": \"rejectedSome\",\n        \"setPath\": true,\n        \"maxAge\": 31536000\n      }\n    ],\n    \"consentedAll\": false,\n    \"dateCreated\": \"2023-11-07T14:10:06.932Z\",\n    \"expirationDate\": \"2024-11-06T14:10:06.932Z\",\n    \"rejectedAll\": false,\n    \"rejectedCategories\": [\n      \"608bae685461ff11a2c28658\"\n    ],\n    \"rejectedVendors\": [\n      \"5dcb0978fcf9312bfc52ae91\"\n    ],\n    \"signedLspa\": false,\n    \"status\": \"rejectedSome\",\n    \"uspstring\": \"1YYN\",\n    \"uuid\": \"7014c66c-1316-4644-aa46-3d85292127a1\"\n  },\n  \"expirationDate\": \"2024-11-06T14:10:06.932Z\"\n}",
  "IABTCF_PurposeOneTreatment": 0,
  "sp.gdpr.key.expiration.date": "2024-11-06T14:04:38.700Z",
  "sp.ccpa.consentUUID": "7014c66c-1316-4644-aa46-3d85292127a1",
  "IABTCF_PublisherConsent": "0000000000",
  "IABTCF_gdprApplies": 1,
  "IABTCF_PublisherRestrictions4": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "IABTCF_PublisherRestrictions2": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "sp.key.messages.v7.local.state": "{\n  \"gdpr\": {\n    \"mmsCookies\": [\n      \"_sp_v1_ss=1:H4sIAAAAAAAAAItWqo5RKimOUbLKK83J0YlRSkVil4AlqmtrlXSoqiwWACMYp9h2AAAA\"\n    ],\n    \"propertyId\": 16893,\n    \"messageId\": 0\n  },\n  \"ccpa\": {\n    \"mmsCookies\": [\n      \"_sp_v1_ss=1:H4sIAAAAAAAAAItWqo5RKimOUbLKK83J0YlRSkVil4AlqmtrlWIBBrfP6SgAAAA%3D\"\n    ],\n    \"propertyId\": 16893,\n    \"messageId\": 0\n  }\n}",
  "IABTCF_PublisherRestrictions7": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "IABTCF_VendorLegitimateInterests": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "IABTCF_PublisherCustomPurposesConsents": "0000000000",
  "IABTCF_PolicyVersion": 4,
  "sp.ccpa.key.expiration.date": "2024-11-06T14:04:38.746Z",
  "IABTCF_AddtlConsent": "1~899",
  "IABTCF_PurposeLegitimateInterests": "0100000000",
  "IABTCF_CmpSdkID": 6,
  "IABTCF_CmpSdkVersion": 2,
  "IABTCF_PublisherCustomPurposesLegitimateInterests": "0000000000",
  "sp.key.meta.data": "{\n  \"ccpa\": {\n    \"applies\": true,\n    \"sampleRate\": 1.0\n  },\n  \"gdpr\": {\n    \"additionsChangeDate\": \"2023-09-19T13:53:18.421Z\",\n    \"applies\": true,\n    \"getMessageAlways\": false,\n    \"_id\": \"608badf1a22863112f750a18\",\n    \"legalBasisChangeDate\": \"2021-06-30T15:52:45.117Z\",\n    \"version\": 47,\n    \"sampleRate\": 1.0\n  }\n}",
  "sp.gdpr.key.sampling.result": true,
  "sp.ccpa.key.sampling.result": true,
  "IABUSPrivacy_String": "1YYN",
  "IABTCF_VendorConsents": "000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "IABTCF_PublisherCC": "DE",
  "IABTCF_SpecialFeaturesOptIns": "00",
  "IABTCF_PublisherLegitimateInterests": "0000000000",
  "IABTCF_PublisherRestrictions10": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "IABTCF_TCString": "CP03uwAP03uwAAGABBENAYEgAKCAAEAAAAYgASAAAAAAQAAAAhAAIAJA.YAAAAAAAAAAA",
  "IABTCF_UseNonStandardTexts": 0,
  "sp.gdpr.consentUUID": "9bf53e38-ab5d-4735-9cab-2158eaa48a8c_25",
  "sp.key.localDataVersion": 1,
  "IABTCF_PurposeConsents": "1010000010",
  "sp.key.config.propertyId": 16893,
  "sp.gdpr.key.consent.status": "{\n  \"categories\": [\n    \"608bad95d08d3112188e0e29\",\n    \"608bad95d08d3112188e0e36\",\n    \"608bad96d08d3112188e0e59\",\n    \"60b65857619abe242bed971e\"\n  ],\n  \"consentedToAll\": false,\n  \"legIntCategories\": [\n    \"608bad95d08d3112188e0e2f\"\n  ],\n  \"legIntVendors\": [\n    \"5f1b2fbeb8e05c306f2a1eb9\"\n  ],\n  \"rejectedAny\": true,\n  \"specialFeatures\": [\n  ],\n  \"vendors\": [\n    \"5f1b2fbeb8e05c306f2a1eb9\",\n    \"5ff4d000a228633ac048be41\",\n    \"5e7ced57b8e05c485246cce0\"\n  ],\n  \"addtlConsent\": \"1~899\",\n  \"consentStatus\": {\n    \"consentedAll\": false,\n    \"consentedToAny\": true,\n    \"granularStatus\": {\n      \"defaultConsent\": false,\n      \"previousOptInAll\": false,\n      \"purposeConsent\": \"SOME\",\n      \"purposeLegInt\": \"ALL\",\n      \"vendorConsent\": \"ALL\",\n      \"vendorLegInt\": \"ALL\"\n    },\n    \"hasConsentData\": true,\n    \"rejectedAny\": true,\n    \"rejectedLI\": false\n  },\n  \"dateCreated\": \"2023-11-07T14:09:57.392Z\",\n  \"euconsent\": \"CP03uwAP03uwAAGABBENAYEgAKCAAEAAAAYgASAAAAAAQAAAAhAAIAJA.YAAAAAAAAAAA\",\n  \"grants\": {\n    \"5e7ced57b8e05c485246cce0\": {\n      \"vendorGrant\": true,\n      \"purposeGrants\": {\n        \"608bad95d08d3112188e0e29\": true,\n        \"608bad95d08d3112188e0e36\": true,\n        \"608bad96d08d3112188e0e59\": true,\n        \"60b65857619abe242bed971e\": true\n      }\n    },\n    \"5f1b2fbeb8e05c306f2a1eb9\": {\n      \"vendorGrant\": true,\n      \"purposeGrants\": {\n        \"608bad95d08d3112188e0e29\": true,\n        \"608bad95d08d3112188e0e2f\": true\n      }\n    },\n    \"5ff4d000a228633ac048be41\": {\n      \"vendorGrant\": false,\n      \"purposeGrants\": {\n        \"608bad95d08d3112188e0e2f\": false,\n        \"608bad95d08d3112188e0e36\": true\n      }\n    }\n  },\n  \"TCData\": {\n    \"IABTCF_AddtlConsent\": \"1~899\",\n    \"IABTCF_CmpSdkID\": 6,\n    \"IABTCF_CmpSdkVersion\": 2,\n    \"IABTCF_PolicyVersion\": 4,\n    \"IABTCF_PublisherCC\": \"DE\",\n    \"IABTCF_PurposeOneTreatment\": 0,\n    \"IABTCF_UseNonStandardTexts\": 0,\n    \"IABTCF_TCString\": \"CP03uwAP03uwAAGABBENAYEgAKCAAEAAAAYgASAAAAAAQAAAAhAAIAJA.YAAAAAAAAAAA\",\n    \"IABTCF_VendorConsents\": \"000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\",\n    \"IABTCF_VendorLegitimateInterests\": \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\",\n    \"IABTCF_PurposeConsents\": \"1010000010\",\n    \"IABTCF_PurposeLegitimateInterests\": \"0100000000\",\n    \"IABTCF_SpecialFeaturesOptIns\": \"00\",\n    \"IABTCF_PublisherConsent\": \"0000000000\",\n    \"IABTCF_PublisherLegitimateInterests\": \"0000000000\",\n    \"IABTCF_PublisherCustomPurposesConsents\": \"0000000000\",\n    \"IABTCF_PublisherCustomPurposesLegitimateInterests\": \"0000000000\",\n    \"IABTCF_gdprApplies\": 1,\n    \"IABTCF_PublisherRestrictions2\": \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"\n  },\n  \"uuid\": \"9bf53e38-ab5d-4735-9cab-2158eaa48a8c_25\",\n  \"webConsentPayload\": {\n    \"actions\": [\n    ],\n    \"addtlConsent\": \"1~899\",\n    \"cookies\": [\n      {\n        \"key\": \"consentUUID\",\n        \"value\": \"9bf53e38-ab5d-4735-9cab-2158eaa48a8c_25\",\n        \"maxAge\": 31536000,\n        \"shareRootDomain\": false\n      },\n      {\n        \"key\": \"consentDate\",\n        \"value\": \"2023-11-07T14:09:57.392Z\",\n        \"maxAge\": 31536000,\n        \"shareRootDomain\": false\n      }\n    ],\n    \"consentStatus\": {\n      \"rejectedAny\": true,\n      \"rejectedLI\": false,\n      \"consentedAll\": false,\n      \"granularStatus\": {\n        \"vendorConsent\": \"ALL\",\n        \"vendorLegInt\": \"ALL\",\n        \"purposeConsent\": \"SOME\",\n        \"purposeLegInt\": \"ALL\",\n        \"previousOptInAll\": false,\n        \"defaultConsent\": false\n      },\n      \"hasConsentData\": true,\n      \"consentedToAny\": true\n    },\n    \"customVendorsResponse\": {\n      \"consentedVendors\": [\n        {\n          \"_id\": \"5f1b2fbeb8e05c306f2a1eb9\",\n          \"name\": \"QuarticON\",\n          \"vendorType\": \"CUSTOM\"\n        },\n        {\n          \"_id\": \"5ff4d000a228633ac048be41\",\n          \"name\": \"Game Accounts\",\n          \"vendorType\": \"CUSTOM\"\n        }\n      ],\n      \"consentedPurposes\": [\n        {\n          \"_id\": \"608bad95d08d3112188e0e29\",\n          \"name\": \"Store and/or access information on a device\"\n        },\n        {\n          \"_id\": \"608bad95d08d3112188e0e36\",\n          \"name\": \"Create profiles for personalised advertising\"\n        },\n        {\n          \"_id\": \"608bad96d08d3112188e0e59\",\n          \"name\": \"Understand audiences through statistics or combinations of data from different sources\"\n        },\n        {\n          \"_id\": \"60b65857619abe242bed971e\",\n          \"name\": \"Our Custom Purpose\"\n        }\n      ],\n      \"legIntPurposes\": [\n        {\n          \"_id\": \"608bad95d08d3112188e0e2f\",\n          \"name\": \"Use limited data to select advertising\"\n        }\n      ]\n    },\n    \"dateCreated\": \"2023-11-07T14:09:57.392Z\",\n    \"expirationDate\": \"2024-11-06T14:09:57.391Z\",\n    \"euconsent\": \"CP03uwAP03uwAAGABBENAYEgAKCAAEAAAAYgASAAAAAAQAAAAhAAIAJA.YAAAAAAAAAAA\",\n    \"grants\": {\n      \"5e7ced57b8e05c485246cce0\": {\n        \"vendorGrant\": true,\n        \"purposeGrants\": {\n          \"608bad95d08d3112188e0e29\": true,\n          \"608bad95d08d3112188e0e36\": true,\n          \"608bad96d08d3112188e0e59\": true,\n          \"60b65857619abe242bed971e\": true\n        }\n      },\n      \"5f1b2fbeb8e05c306f2a1eb9\": {\n        \"vendorGrant\": true,\n        \"purposeGrants\": {\n          \"608bad95d08d3112188e0e29\": true,\n          \"608bad95d08d3112188e0e2f\": true\n        }\n      },\n      \"5ff4d000a228633ac048be41\": {\n        \"vendorGrant\": false,\n        \"purposeGrants\": {\n          \"608bad95d08d3112188e0e2f\": false,\n          \"608bad95d08d3112188e0e36\": true\n        }\n      }\n    },\n    \"uuid\": \"9bf53e38-ab5d-4735-9cab-2158eaa48a8c_25\"\n  },\n  \"expirationDate\": \"2024-11-06T14:09:57.391Z\"\n}",
  "sp.key.messages.v7.nonKeyedLocalState": "{\n  \"ccpa\": {\n    \"_sp_v1_data\": \"585620\",\n    \"_sp_v1_p\": \"461\"\n  },\n  \"gdpr\": {\n    \"_sp_v1_data\": \"696784\",\n    \"_sp_v1_p\": \"205\"\n  }\n}"
}
`

Reply:
ok, let me sand this json via slack 😅

Reply:
great, thank you for that @carmelo-iriti